### PR TITLE
Proportionally split the payment processor fees when collecting fees on top

### PR DESCRIPTION
**Pull request by kewitz**

### CODE REVIEW

**Comment by Mr1262** [comment]
fix/fees-on-top-processor-fees

File: server/models/Transaction.js [hunk 1]
```diff
@@ -498,6 +505,8 @@ export default (Sequelize, DataTypes) => {
     await Transaction.createDoubleEntry(donationTransaction);
 
     // Remove fees from main transaction
+    transaction.paymentProcessorFeeInHostCurrency =
+      transaction.paymentProcessorFeeInHostCurrency - paymentProcessorFeeInHostCurrency;
```
Comment (znarf) [inline server/models/Transaction.js:509]: `paymentProcessorFeeInHostCurrency - paymentProcessorFeeInHostCurrency;` is difficult to read.

Can you consider renaming variables?

**Comment by kewitz** [comment]
@Mr1262 ?

**Comment by znarf** [comment]
I'm good with it, would be great to have a second review from @Betree.

File: server/models/Transaction.js [hunk 2]
```diff
@@ -476,6 +476,13 @@ export default (Sequelize, DataTypes) => {
       return;
     }
 
+    // Calculate the paymentProcessorFee proportional to the feeOnTop amount
+    const feeOnTopPaymentProcessorFee = toNegative(
+      Math.round(
+        transaction.paymentProcessorFeeInHostCurrency *
+          Math.abs(transaction.platformFeeInHostCurrency / transaction.amountInHostCurrency),
+      ),
+    );
```
Comment (Betree) [inline server/models/Transaction.js:485]: When doing math in code, I personally find that using as many variables as possibles improves clarity by naming things.  But that's a personal preference

```suggestion
    const feeOnTopPercent = Math.abs(transaction.platformFeeInHostCurrency / transaction.amountInHostCurrency);
    const feeOnTopPaymentProcessorFee = toNegative(
      Math.round(transaction.paymentProcessorFeeInHostCurrency * feeOnTopPercent),
    );
```

**Pull request merged**